src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -3422,12 +3418,44 @@ def convert_func_to_agg_func_info(
         )
         # the pandas label and quoted identifier generated for each result column
         # after aggregation will be used as new pandas label and quoted identifiers.
-        new_data_column_pandas_labels = [
-            col_agg_op.agg_pandas_label for col_agg_op in agg_col_ops
-        ]
-        new_data_column_quoted_identifier = [
-            col_agg_op.agg_snowflake_quoted_identifier for col_agg_op in agg_col_ops
-        ]
+        new_data_column_pandas_labels = []
+        new_data_column_quoted_identifiers = []
+        new_data_column_snowpark_pandas_types = []
+        for col_agg_op in agg_col_ops:
+            new_data_column_pandas_labels.append(col_agg_op.agg_pandas_label)
+            new_data_column_quoted_identifiers.append(
+                col_agg_op.agg_snowflake_quoted_identifier
+            )
+            if agg_func in (
+                "min",
+                "max",
+                "sum",
+                "mean",
+                "median",
+                "std",
+                "first",
+                "last",
+            ):
+                new_data_column_snowpark_pandas_types.append(
+                    col_agg_op.data_type
+                    if isinstance(col_agg_op.data_type, SnowparkPandasType)
+                    else None
+                )
+            elif agg_func in (
+                "any",
+                "all",
+                "count",
+                "idxmax",
+                "idxmin",
+                "size",
+                "nunique",
+            ):
+                # In the case where the aggregation overrides the type of the output data column
+                # (e.g. any always returns boolean data columns), set the output Snowpark pandas type to None
+                new_data_column_snowpark_pandas_types = None  # type: ignore
+            else:
+                self._raise_not_implemented_error_for_timedelta()
```

Comment:
if instead we can list the not implemented agg_func that will be better I guess. Not sure is it feasible though.

Reply:
We could in theory do it, but I would rather have a catch-all with the `else` statement here. This was the original behavior before supporting Timedelta anyhow so I think it's better to default to NotImplementedError for Timedelta.